runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -94,15 +95,34 @@ end
 ---
 ---@private
 ---@return STTokenRange[]
-local function tokens_to_ranges(data, bufnr, client)
+local function tokens_to_ranges(data, bufnr, client, request)
   local legend = client.server_capabilities.semanticTokensProvider.legend
   local token_types = legend.tokenTypes
   local token_modifiers = legend.tokenModifiers
   local ranges = {}
 
+  local start = uv.hrtime()
+  local ns_to_ms = 0.000001
+  local co = coroutine.running()
+
   local line
   local start_char = 0
   for i = 1, #data, 5 do
```

Comment:
Unrelated to this PR, but what is this `5` supposed to mean?

Reply:
It's part of the LSP semantic token spec. Every 5 values in the array relate to a token.